cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -0,0 +1,177 @@
+                val (moveCashCommand, cashOutputState) = it.state.data.withNewOwner(iouState.lender)
+                txBuilder.addInputState(it)
+                txBuilder.addOutputState(cashOutputState, it.state.contract, it.state.notary)
+                txBuilder.addCommand(moveCashCommand, it.state.data.owner.owningKey)
+            }
+
+            progressTracker.currentStep = VERIFYING_TRANSACTION
+            // Verify that the transaction is valid.
+            txBuilder.verify(serviceHub)
+
+            progressTracker.currentStep = SIGNING_TRANSACTION
+            // Sign the transaction (must be by a borrower)
+            if (!iouState.borrower.equals(ourIdentity)) {
+                throw IllegalArgumentException("Borrower must start the flow")
+            }
+            val signedTx = serviceHub.signInitialTransaction(txBuilder)
+
+            progressTracker.currentStep = INITIALISE_OTHER_PARTY_FLOW
+            // Send the state to the counterparty, and receive it back with their signature.
+            val lenderPartySession = initiateFlow(iouState.lender)
+
+            // send/receive part is just for experiment
+            val confirmPayback = lenderPartySession
+                    .sendAndReceive<Boolean>("ready for receiving payback?")
+                    .unwrap { it }
+
+            require(confirmPayback) { "Oops, lender failed to be ready" }
+
+            progressTracker.currentStep = PAYBACK_COMMUNICATION
+            lenderPartySession.send(iouState.value)
+
+            progressTracker.currentStep = GATHERING_SIGS
+            val fullySignedTx = subFlow(CollectSignaturesFlow(
+                    signedTx,
+                    setOf(lenderPartySession),
+                    GATHERING_SIGS.childProgressTracker())
+            )
+
+            progressTracker.currentStep = FINALISING_TRANSACTION
+            // Notarise and record the transaction in both parties' vaults.
+            return subFlow(FinalityFlow(fullySignedTx, setOf(lenderPartySession), FINALISING_TRANSACTION.childProgressTracker()))
+        }
+    }
+
+    @InitiatedBy(Initiator::class)
+    class Acceptor(val borrowerPartySession: FlowSession) : FlowLogic<SignedTransaction>() {
+        @Suspendable
+        override fun call(): SignedTransaction {
+            borrowerPartySession.receive<String>().unwrap { it }
+            //send confirmation about being ready to receive payback
+            borrowerPartySession.send(true)
+
+            // just exercise
+            val receivedPayback: Int = borrowerPartySession.receive<Int>().unwrap { it }
+
+            val signTransactionFlow = object : SignTransactionFlow(borrowerPartySession) {
+                override fun checkTransaction(stx: SignedTransaction) = requireThat {
+                    val iouState = serviceHub.toStateAndRef<IOUState>(stx.tx.inputs.first()).state.data
+                    // val inCashStates = serviceHub.toStateAndRef<CashState>(stx.tx.inputs.last()).state.data
```

Comment:
You should also check the input cash states because perhaps the borrower is trying to pass to the lender states that are owned by the lender 😂 
And of course, it is a list you should care about. There is a difficulty here, but try on your own first.

Reply:
ok, I done it... commit soon :)